packages/core/src/components/external/Toast/Toast.tsx

```diff
@@ -0,0 +1,151 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2023 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, {useMemo, useState} from 'react';
+import {StyleSheet, View} from 'react-native';
+import {
+  default as ReactNativeToast,
+  BaseToast,
+  ErrorToast,
+} from 'react-native-toast-message';
+import {
+  Alert,
+  Icon,
+  Text,
+  ThemeColors,
+  useThemeColor,
+} from '@axelor/aos-mobile-ui';
+import {useTranslator} from '../../../i18n';
+import toastProvider from './ToastProvider';
+
+const Toast = () => {
+  const Colors = useThemeColor();
+  const I18n = useTranslator();
+
+  const toastData = toastProvider.getData();
+
+  const isError = useMemo(() => toastData.type === 'error', [toastData]);
+
+  const toastColor = useMemo(
+    () =>
+      isError ? Colors.errorColor.background : Colors.successColor.background,
+    [Colors, isError],
+  );
+
+  const styles = useMemo(
+    () => getStyles(Colors, toastColor),
+    [Colors, toastColor],
+  );
+
+  const [isAlertVisible, setIsAlertVisible] = useState(false);
+
+  const toastConfig = {
+    success: (props: any) => (
+      <BaseToast
+        {...props}
+        style={styles.toast}
+        contentContainerStyle={styles.toastContent}
+        text1Style={styles.title}
+        text2Style={styles.detail}
+        text2NumberOfLines={3}
+      />
+    ),
+    error: (props: any) => (
+      <ErrorToast
+        {...props}
+        style={styles.toast}
+        contentContainerStyle={styles.toastContent}
+        text1Style={styles.title}
+        text2Style={styles.detail}
+        text2NumberOfLines={3}
+      />
+    ),
+  };
```

Comment:
I think there is a refresh issue on border color. When defining toast config, border color should not depends on useMemo. It should be clearly written : success = success color, eror = error color
At the moment, toast appear as the last toast color is border color is updated when you click on the toast